std/io/ioutil_test.ts

```diff
@@ -98,3 +100,22 @@ Deno.test("copyNWriteAllData", async function (): Promise<void> {
 
   assertEquals(n, size);
 });
+
+Deno.test("testStringReaderEof", async function (): Promise<void> {
+  const r = new StringReader("abc");
```

Comment:
use `stringsReader("abc")`

Reply:
But then it won't test StringReader, which has the bug

Reply:
Ah I see. Somehow this StringReader class was added erroneously. The correct implementation is stringsReader.

Reply:
maybe test with both StringReader and stringsReader(Buffer)?

Reply:
So the StringReader should probably just be removed

Reply:
and replaced with stringsReader once every occurrence